File: VocaDbModel/Domain/Songs/SongVoteRating.cs
namespace VocaDb.Model.Domain.Songs {

	/// <summary>
	/// Rating of a song vote.
	/// Saved in DB as integer - do not change the integer values here.
	/// </summary>
	public enum SongVoteRating {

		Nothing		= 0,

		Dislike		= 1,

		Like		= 3,

		Favorite	= 5

	}
}

File: VocaDbWeb/Scripts/Models/SongVoteRating.ts
// Song vote values.
// Corresponds to the enum SongVoteRating.
enum SongVoteRating {
  Nothing = 0,
  Like = 3,
  Favorite = 5,
}

export default SongVoteRating;

export function parseSongVoteRating(rating: string): SongVoteRating {
  switch (rating) {
    case 'Nothing':
      return SongVoteRating.Nothing;
    case 'Like':
      return SongVoteRating.Like;
    case 'Favorite':
      return SongVoteRating.Favorite;
    default:
      return undefined!;
  }
}
